### PR TITLE
Added auth0 prompt custom text

### PR DESCRIPTION
**Pull request by jluiz20**
### Proposed Changes

* Introduces `auth0_prompt_custom_text`

#### Acceptance Test Output
<details>
  <summary>Click to expand test output</summary>
  
```
$ make testacc TESTS=TestAccPromptCustomText
==> Checking that code complies with gofmt requirements...
?       github.com/alexkappa/terraform-provider-auth0   [no test files]
=== RUN   TestAccPromptCustomText
--- PASS: TestAccPromptCustomText (4.68s)
PASS
coverage: 5.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0     7.574s  coverage: 5.0% of statements
?       github.com/alexkappa/terraform-provider-auth0/auth0/internal/debug      [no test files]
?       github.com/alexkappa/terraform-provider-auth0/auth0/internal/digitalocean       [no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/hash       0.390s  coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/random     0.424s  coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/validation 0.610s  coverage: 0.0% of statements [no tests to run]
?       github.com/alexkappa/terraform-provider-auth0/version   [no test files]
```
</details>

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by jluiz20** [comment]
This is an attempt to add the prompt custom texts to the terraform resource. This is my first change to a terraform resource project, so any feedback is very welcome.

The challenge that we have in the custom text is that they don't seem to have an ID (at least none is returned on the API). One way to create an ID is using the prompt type and the language. I used this to create the ID, but I am not sure what is default pattern is for this kind of situation.

If you have any ideas for improvements or another way to do it, please provide feedback.

PS: I have tested the code, either using the test and also using the compiled file and it works fine.

**Comment by kiranms-newpage** [comment]
@sergiughf We have been waiting for this feature since long. Can you please review this and merge and release if all looks okay to you?

**Comment by sergiughf** [comment]
Hey folks, I'll take a look at this today. 🖖🏻

**Comment by sergiughf** [comment]
Quick update that I started taking a look at this but haven't yet finished running through all the manual tests I wanted to perform. I'll give another update later in the day.

**Comment by sergiughf** [comment]
Hey @jluiz20, thanks a lot for the contribution! The PR was in a pretty good shape, I only pushed a couple of small improvements that we can go over together if you have any questions. 

> The challenge that we have in the custom text is that they don't seem to have an ID (at least none is returned on the API). One way to create an ID is using the prompt type and the language. I used this to create the ID, but I am not sure what is default pattern is for this kind of situation.

Your assumption is correct, the `prompty_type` + `language` forms a unique pair that we can consider as the ID for each one of them. 

The only thing I feel is missing from this PR is to add functionality on the `deletePromptCustomText` IMO ideally this should reset the body to `{}` instead of not doing anything. Wdyt?

File: auth0/resource_auth0_prompt_custom_text.go [hunk 1]
```diff
@@ -14,48 +14,55 @@ import (
...
-				}, false),
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(availableLanguages, false),
 			},
 			"body": {
 				Type:             schema.TypeString,
 				Required:         true,
-				ValidateFunc:     validation.ValidateJsonString,
+				ValidateFunc:     validation.StringIsJSON,
```
Comment (sergiughf) [inline auth0/resource_auth0_prompt_custom_text.go:56]: This was changed as `ValidateJsonString` is deprecated.

File: auth0/resource_auth0_prompt_custom_text.go [hunk 2]
```diff
@@ -14,48 +14,55 @@ import (
 	"gopkg.in/auth0.v5/management"
 )
 
-func newPromptCustomText() *schema.Resource {
+var (
+	availablePrompts = []string{
```
Comment (sergiughf) [inline auth0/resource_auth0_prompt_custom_text.go:18]: Created these vars for holding all the prompts and countries so it's a bit more explicit what they are.

File: auth0/resource_auth0_prompt_custom_text_test.go [hunk 3]
```diff
@@ -19,49 +18,57 @@ func TestAccPromptCustomText(t *testing.T) {
...
   language = "en"
-  body = <<EOF
-{
-"login": {
-	"alertListTitle": "Alerts",
-	"buttonText": "Continue",
-	"emailPlaceholder": "Email address"}
-}
-EOF
+  body = jsonencode(
```
Comment (sergiughf) [inline auth0/resource_auth0_prompt_custom_text_test.go:48]: I changed this as well to use jsonencode so we can write and format the json payload as an object in HCL.

File: auth0/resource_auth0_prompt_custom_text.go (hunk 2 above)
Comment (jluiz20) [reply auth0/resource_auth0_prompt_custom_text.go:18]: Nice idea!

File: auth0/resource_auth0_prompt_custom_text_test.go (hunk 3 above)
Comment (jluiz20) [reply auth0/resource_auth0_prompt_custom_text_test.go:48]: nice. didn't know about that

**Comment by jluiz20** [comment]
Hi @sergiughf. Thanks, very happy to contribute to the provider that I am using :D 
Thank you for the improvements.

your suggestion about the missing reset for the body is good. I have already made the changes. 

Thank you for reviewing!

**Comment by jluiz20** [comment]
@sergiughf, just added on more change, see if my thinking is right:

```
func deletePromptCustomText(d *schema.ResourceData, m interface{}) error {
	d.Set("body", "{}")

	updatePromptCustomText(d, m)

	d.SetId("")
	return nil
}
```

What I did was add the `updatePromptCustomText(d,m)` after resetting the body. As we don't have an API for deletion, this `{} ` clear the custom texts.

What do you think?

**Comment by sergiughf** [comment]
Yeah that makes sense! Thanks @jluiz20! ⭐  I tested locally and everything seems to work, although could I ask you to please not ignore the error on the delete when you call `updatePromptCustomText(d, m)`? 🙏🏻

File: auth0/resource_auth0_prompt_custom_text.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+		}
+	}
+
+	return readPromptCustomText(d, m)
+}
+
+func deletePromptCustomText(d *schema.ResourceData, m interface{}) error {
+	d.Set("body", "{}")
+
+	updatePromptCustomText(d, m)
```
Comment (sergiughf) [inline auth0/resource_auth0_prompt_custom_text.go:125]: Let's please make sure we don't ignore this error:) 
```suggestion
	if err := updatePromptCustomText(d, m); err != nil {
		return err
	}
```

**Comment by jluiz20** [comment]
off course. My bad for ignoring the error. I am pretty new to golang 😬

**Comment by sergiughf** [comment]
For being pretty new to golang you did a great job @jluiz20 🏆 

Thanks a lot for the contribution, this is going to help a lot of folks! 

I'll merge this first thing in the morning.

**Review by florian-bellotti — APPROVED** [review]

**Comment by sergiughf** [comment]
Hey @jluiz20 👋🏻  I added a new commit to improve the error handling on importing the custom text https://github.com/alexkappa/terraform-provider-auth0/pull/497/commits/f017f3ec2156148e7b9c2799bab0f12e0962fa85 and rebased to push the changelog for the new version.

**Review by sergiughf — APPROVED** [review]

**Pull request merged**